assets/pages/new-contributions-landing/contributionsLandingActions.js

```diff
@@ -127,6 +132,47 @@ const setThirdPartyPaymentLibrary =
 
 const setPayPalHasLoaded = (): Action => ({ type: 'SET_PAYPAL_HAS_LOADED' });
 
+const setisSignInRequired = (isSignInRequired: boolean): Action => ({
+  type: 'SET_SIGN_IN_REQUIRED',
+  isSignInRequired,
+});
+
+const setIdentityRequestPending = (isIdentityRequestPending: boolean): Action => ({
+  type: 'SET_IDENTITY_REQUEST_PENDING',
+  isIdentityRequestPending,
+});
+
+const setLastIdentityResponse = (lastIdentityResponse: IdentityResponse): Action => ({
+  type: 'SET_LAST_IDENTITY_RESPONSE',
+  lastIdentityResponse,
+});
+
+const checkIfEmailHasPassword = (email: string) =>
+  (dispatch: Dispatch<Action>, getState: () => State): void => {
+    const state = getState();
+    dispatch(setIdentityRequestPending(true));
+    fetchJson(
+      `/dummy/${encodeURIComponent(email)}`,
+      getRequestOptions('same-origin', state.page.csrf),
+    ).then(({ userType }) => {
+      dispatch(setIdentityRequestPending(false));
+      if (typeof userType !== 'string') {
+        throw new Error('userType string was not present in response');
+      }
+      dispatch(setLastIdentityResponse('success'));
```

Comment:
Do we need a way of ensuring that the last response from identity is the response from the last request to identity?